warehouse/packaging/models.py

```diff
@@ -268,6 +270,37 @@ def latest_version(self):
             .first()
         )
 
+    @property
+    def latest_stable_version(self):
+        # Supply the latest stable version. If no stable versions are
+        # available, return None.
+        return (
+            orm.object_session(self)
+            .query(Release.version, Release.created, Release.is_prerelease)
+            .filter(
+                Release.project == self,
+                Release.yanked.is_(False),
+                Release.is_prerelease.is_(False),
+            )
+            .order_by(Release._pypi_ordering.desc())
+            .first()
+        )
+
+    @property
+    def latest_unstable_version(self):
+        # Supply the latest available version, regardless of pre-release status.
+        return (
+            orm.object_session(self)
+            .query(Release.version, Release.created, Release.is_prerelease)
+            .filter(
+                Release.project == self,
+                Release.yanked.is_(False),
+                Release.is_prerelease is not None,
```

Comment:
I might misread it but this seems to contradict with the docstring: If I'm not mistaken, this selects only prereleases, where we'd want a stable if it's the most recent.

Reply:
I think this is ok -- the filter is `Release.is_prerelease is not None`, which is just a (possibly unnecessary) filter against a malformed table row; as opposed to `Release.is_prerelease is True` (or simply `Release.is_prerelease`), which I think would be a filter that would return only pre-releases.